cpp/test/neighbors/ann_cagra.cuh

```diff
@@ -1006,29 +1032,167 @@ inline std::vector<AnnCagraInputs> generate_inputs()
+                                                   {0},  // team_size
+                                                   {64},
+                                                   {1},
+                                                   {cuvs::distance::DistanceType::L2Expanded},
+                                                   {false},
+                                                   {false},
+                                                   {0.995});
+  for (auto input : inputs2) {
+    input.non_owning_memory_buffer_flag = true;
+    inputs.push_back(input);
+  }
+
+  return inputs;
+}
+
+inline std::vector<AnnCagraInputs> generate_addnode_inputs()
+{
+  // changing dim
+  std::vector<AnnCagraInputs> inputs =
+    raft::util::itertools::product<AnnCagraInputs>({100},
+                                                   {1000},
+                                                   {1, 8, 17, 64, 128, 137, 512, 1024},  // dim
+                                                   {16},                                 // k
+                                                   {graph_build_algo::NN_DESCENT},
+                                                   {search_algo::AUTO},
+                                                   {10},
+                                                   {0},
+                                                   {64},
+                                                   {1},
+                                                   {cuvs::distance::DistanceType::L2Expanded},
+                                                   {false},
+                                                   {true},
+                                                   {0.995});
+
+  // testing host and device datasets
+  auto inputs2 =
+    raft::util::itertools::product<AnnCagraInputs>({100},
+                                                   {10000},
+                                                   {32},
+                                                   {10},
+                                                   {graph_build_algo::AUTO},
+                                                   {search_algo::AUTO},
+                                                   {10},
+                                                   {0},  // team_size
+                                                   {64},
+                                                   {1},
+                                                   {cuvs::distance::DistanceType::L2Expanded},
+                                                   {false, true},
+                                                   {false},
+                                                   {0.985});
+  inputs.insert(inputs.end(), inputs2.begin(), inputs2.end());
+
+  // a few PQ configurations
   inputs2 = raft::util::itertools::product<AnnCagraInputs>(
     {100},
-    {1000},
-    {1, 3, 5, 7, 8, 17, 64, 128, 137, 192, 256, 512, 619, 1024},  // dim
-    {10},
+    {10000},
+    {192, 1024},  // dim
```

Comment:
When testing AddNode, I assumed it is fine to limit to a smaller set of dimensions.